REPL crash with tab completion - invalid prefix ExprType

## Compiler version

3.0.1 repl

## Minimized code

```Scala
// command 1
import quoted.*
```

```Scala
// command 2
def fooImpl(using Quotes): Expr[Int] = {
  import quotes.reflect.*
  TypeRepr.of[Int].<tab>
}
```

## Output (click arrow to expand)
<details>

```scala
Exception in thread "main" java.lang.AssertionError: assertion failed: invalid prefix ExprType(TypeRef(TermRef(TermRef(NoPrefix,val x$1),val reflect),trait ConstantTypeMethods))
	at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:8)
	at dotty.tools.dotc.core.Types$NamedType.<init>(Types.scala:2025)
	at dotty.tools.dotc.core.Types$TermRef.<init>(Types.scala:2567)
	at dotty.tools.dotc.core.Types$CachedTermRef.<init>(Types.scala:2646)
	at dotty.tools.dotc.core.Uniques$NamedTypeUniques.newType$1(Uniques.scala:42)
	at dotty.tools.dotc.core.Uniques$NamedTypeUniques.enterIfNew(Uniques.scala:53)
	at dotty.tools.dotc.core.Types$TermRef$.apply(Types.scala:2697)
	at dotty.tools.dotc.interactive.Completion$$anon$3.applyOrElse(Completion.scala:326)
	at dotty.tools.dotc.interactive.Completion$$anon$3.applyOrElse(Completion.scala:326)
	at scala.collection.immutable.List.collect(List.scala:267)
	at scala.collection.immutable.List.collect(List.scala:79)
	at dotty.tools.dotc.interactive.Completion$Completer.extractMemberExtensionMethods$2$$anonfun$1(Completion.scala:326)
	at scala.collection.immutable.List.flatMap(List.scala:293)
	at scala.collection.immutable.List.flatMap(List.scala:79)
	at dotty.tools.dotc.interactive.Completion$Completer.extractMemberExtensionMethods$1(Completion.scala:326)
	at dotty.tools.dotc.interactive.Completion$Completer.extensionCompletions(Completion.scala:347)
	at dotty.tools.dotc.interactive.Completion$Completer.selectionCompletions(Completion.scala:243)
	at dotty.tools.dotc.interactive.Completion$.computeCompletions(Completion.scala:115)
	at dotty.tools.dotc.interactive.Completion$.completions(Completion.scala:49)
	at dotty.tools.repl.ReplDriver.completions$$anonfun$1(ReplDriver.scala:203)
	at scala.util.Either.map(Either.scala:382)
	at dotty.tools.repl.ReplDriver.completions(ReplDriver.scala:204)
	at dotty.tools.repl.ReplDriver.$anonfun$3(ReplDriver.scala:128)
	at org.jline.reader.impl.LineReaderImpl.doComplete(LineReaderImpl.java:4412)
	at org.jline.reader.impl.LineReaderImpl.doComplete(LineReaderImpl.java:4378)
	at org.jline.reader.impl.LineReaderImpl.expandOrComplete(LineReaderImpl.java:4317)
	at org.jline.reader.impl.LineReaderImpl$1.apply(LineReaderImpl.java:3809)
	at org.jline.reader.impl.LineReaderImpl.readLine(LineReaderImpl.java:679)
	at org.jline.reader.impl.LineReaderImpl.readLine(LineReaderImpl.java:468)
	at dotty.tools.repl.JLineTerminal.readLine(JLineTerminal.scala:72)
	at dotty.tools.repl.ReplDriver.readLine$1(ReplDriver.scala:133)
	at dotty.tools.repl.ReplDriver.loop$1(ReplDriver.scala:143)
	at dotty.tools.repl.ReplDriver.runUntilQuit$$anonfun$1(ReplDriver.scala:148)
	at dotty.tools.repl.ReplDriver.withRedirectedOutput(ReplDriver.scala:167)
	at dotty.tools.repl.ReplDriver.runUntilQuit(ReplDriver.scala:148)
	at dotty.tools.repl.ReplDriver.tryRunning(ReplDriver.scala:114)
	at dotty.tools.repl.Main$.main(Main.scala:6)
	at dotty.tools.repl.Main.main(Main.scala)
```
</details>

--- a/compiler/src/dotty/tools/dotc/interactive/Completion.scala
+++ b/compiler/src/dotty/tools/dotc/interactive/Completion.scala
@@ -300,7 +300,8 @@ object Completion {
               case name: TermName if include(denot, name) => Some((denot, name))
               case _ => None
 
-        types.flatMap { tpe =>
+        types.flatMap { tp =>
+          val tpe = tp.widenExpr
           tpe.membersBasedOnFlags(required = ExtensionMethod, excluded = EmptyFlags)
             .collect { case DenotWithMatchingName(denot, name) => TermRef(tpe, denot.symbol) -> name }
         }

--- a/compiler/test/dotty/tools/repl/TabcompleteTests.scala
+++ b/compiler/test/dotty/tools/repl/TabcompleteTests.scala
@@ -132,4 +132,10 @@ class TabcompleteTests extends ReplTest {
   @Test def i6415 = fromInitialState { implicit s =>
     assertEquals(List("Predef"), tabComplete("object Foo { opaque type T = Pre"))
   }
+
+  @Test def i12600 = fromInitialState { implicit s =>
+    assertEquals(List("select", "show", "simplified"),
+      tabComplete("import quoted.* ; def fooImpl(using Quotes): Expr[Int] = { import quotes.reflect.* ; TypeRepr.of[Int].s"))
+  }
+
 }

--- a/language-server/test/dotty/tools/languageserver/CompletionTest.scala
+++ b/language-server/test/dotty/tools/languageserver/CompletionTest.scala
@@ -842,4 +842,19 @@ class CompletionTest {
           |object Main { "abc".xx${m1} }""".withSource
       .completion(m1, Set())
   }
+
+  @Test def i13365: Unit = {
+    code"""|import scala.quoted._
+        |
+        |object Test {
+        |  def test(using Quotes)(str: String) = {
+        |    import quotes.reflect._
+        |    val msg = Expr(str)
+        |    val printHello = '{ print("sdsd") }
+        |    val tree = printHello.asTerm
+        |    tree.sh${m1}@@ // doesn't work
+        |  }
+        |}""".withSource
+      .completion(m1, Set(("show",Method, "(using x$2: x$1.reflect.Printer[x$1.reflect.Tree]): String")))
+  }
 }

--- a/language-server/test/dotty/tools/languageserver/CompletionTest.scala
+++ b/language-server/test/dotty/tools/languageserver/CompletionTest.scala
@@ -852,7 +852,7 @@ class CompletionTest {
         |    val msg = Expr(str)
         |    val printHello = '{ print("sdsd") }
         |    val tree = printHello.asTerm
-        |    tree.sh${m1}@@ // doesn't work
+        |    tree.sh${m1}
         |  }
         |}""".withSource
       .completion(m1, Set(("show",Method, "(using x$2: x$1.reflect.Printer[x$1.reflect.Tree]): String")))

